# 9.10 练习

(1) 用main()创建一个类，令其掷出try块内的Exception类的一个对象。为Exception的构造器赋予一个字串参数。在catch从句内捕获异常，并打印出字串参数。添加一个finally从句，并打印一条消息，证明自己真正到达那里。

(2) 用extends关键字创建自己的异常类。为这个类写一个构造器，令其采用String参数，并随同String指针把它保存到对象内。写一个方法，令其打印出保存下来的String。创建一个try-catch从句，练习实际操作新异常。

(3) 写一个类，并令一个方法掷出在练习2中创建的类型的一个异常。试着在没有异常规范的前提下编译它，观察编译器会报告什么。接着添加适当的异常规范。在一个try-catch从句中尝试自己的类以及它的异常。

(4) 在第5章，找到调用了Assert.java的两个程序，并修改它们，令其掷出自己的异常类型，而不是打印到System.err。该异常应是扩展了RuntimeException的一个内部类。